lib/foreman_ansible_core/variables_extractor.rb

```diff
@@ -0,0 +1,30 @@
+module ForemanAnsibleCore
+  # Implements the logic needed to read the roles and associated information
+  class VariablesExtractor
+    class << self
+      def extract_variables(role_path)
+        role_files = Dir.glob("#{role_path}/**/*.yml")
```

Comment:
Does it make sense to scan for *.yaml files as well?

But more importantly, why getting variables by scanning the files like this? I plan to test some more, but it seems fragile to me. Variables from defaults/main.yml that are used just in templates are not imported. Wouldn't it be easier just to get them from the files in defaults?

Reply:
Sometimes there is no 'defaults' folder, and we would still want to read the variables in such a case

Reply:
I do not think this is resolved. There are 3 main problems with this approach as far as I can see:

* the variables that are declared in defaults/main.yml and used in templates are not detected

* you are matching more that just variables, [filters](https://docs.ansible.com/ansible/2.5/user_guide/playbooks_filters.html#filters-for-formatting-data) are imported as variables.

* you are importing not just variables of the role, you are importing all the variables that are used in the role. Do we really want to import `ansible_distribution` as a role variable? 

![](https://imgur.com/rDOYdoa.png)


What about the following:

```yaml
tasks:
    - set_fact:
        qpid_port: "{{ lookup('env', 'QPID_SERVICE_PORT') | int }}"
        qpid_host: "{{ lookup('env', 'QPID_SERVICE_HOST') }}"

    - name: "Wait for Qpid"
      wait_for:
        port: "{{ qpid_port }}"
        host: "{{ qpid_host }}"
        timeout: 600

    - name: "Check for qpid queue"
      command: "qpid-config -b {{ qpid_host }}:{{ qpid_port }} queues katello_event_queue"
      register: queue_exist
      ignore_errors: true
```

Should `qpid_port` and `qpid_host` be exported as variables?